presto-main/src/main/java/com/facebook/presto/operator/aggregation/differentialentropy/DifferentialEntropyAggregation.java

```diff
@@ -0,0 +1,143 @@
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.AggregationState;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+
+@AggregationFunction("differential_entropy")
+@Description("Computes differential entropy based on random-variable samples")
+public final class DifferentialEntropyAggregation
+{
+    private DifferentialEntropyAggregation() {}
+
+    @InputFunction
+    public static void input(
+            @AggregationState DifferentialEntropyState state,
+            @SqlType(StandardTypes.BIGINT) long size,
+            @SqlType(StandardTypes.DOUBLE) double sample,
+            @SqlType(StandardTypes.DOUBLE) double weight,
+            @SqlType(StandardTypes.VARCHAR) Slice method,
+            @SqlType(StandardTypes.DOUBLE) double arg0,
+            @SqlType(StandardTypes.DOUBLE) double arg1)
+    {
+        inputImp(state, size, sample, weight, method, Double.valueOf(arg0), Double.valueOf(arg1));
+    }
+
+    @InputFunction
+    public static void input(
+            @AggregationState DifferentialEntropyState state,
+            @SqlType(StandardTypes.BIGINT) long size,
+            @SqlType(StandardTypes.DOUBLE) double sample,
+            @SqlType(StandardTypes.DOUBLE) double weight,
+            @SqlType(StandardTypes.VARCHAR) Slice method)
+    {
+        inputImp(state, size, sample, weight, method, null, null);
+    }
+
+    @InputFunction
+    public static void input(
+            @AggregationState DifferentialEntropyState state,
+            @SqlType(StandardTypes.BIGINT) long size,
+            @SqlType(StandardTypes.DOUBLE) double sample,
+            @SqlType(StandardTypes.DOUBLE) double weight)
+    {
+        inputImp(
+                state,
+                size,
+                sample,
+                weight,
+                null,
+                (Double) null,
+                (Double) null);
+    }
+
+    @InputFunction
+    public static void input(
+            @AggregationState DifferentialEntropyState state,
+            @SqlType(StandardTypes.BIGINT) long size,
+            @SqlType(StandardTypes.DOUBLE) double sample)
+    {
+        inputImp(
+                state,
+                size,
+                sample,
+                null,
+                null,
+                null,
+                null);
+    }
+
+    protected static void inputImp(
+            DifferentialEntropyState state,
+            long size,
+            double sample,
+            Double weight,
+            Slice method,
+            Double arg0,
+            Double arg1)
```

Comment:
We should name the variables

Reply:
There's actually a reason why these are called generically. The general format of differential entropy I suggest is 

size, sample, weight, method, ...

which I think makes sense, as any sketch has a concept of a size, then there's a sample, optionally a weight, optionally a method, and perhaps more parameters. These two parameters here are min and max for histograms, but just don't exist for reservoir sampling, and could mean other things for compressed counting, etc.